msupply-foundation/open-msupply
# 3059 insert outbound return

Fixes #3059

She's a chunky one - sorry 😭 

# 👩🏻‍💻 What does this PR do? 
 

The creating an Outbound Return from an Inbound Shipment (or from the Outbound Return list view) now actually creates a Return! 

Also hooks up the return_reasons query to the return_reasons table 😁 

# 🧪 How has/should this change been tested? 


Automated tests added for the service layer.

To test in the UI, you'll need to populate yourself some return reasons. From wherever you like to run your SQL:

```sql
INSERT INTO return_reason (id, reason, is_active)
VALUES 
('2153c064-7921-41f8-9716-1b91693d1346', 'Damaged', TRUE),
('c81213b1-d28e-4a44-bddc-41a49ddb4117', 'Expired', TRUE),
('f2571df2-d605-4695-8ce5-4e7242ae1ab3', 'Wrong item', TRUE),
('38fe99d1-d84a-4f95-bd29-4938f4128246', 'Other', TRUE);
```

- Create a return from the Outbound Returns list view (this will only create the invoice, invoice lines would be added from the detail view in future)
- Create a return from an Inbound Shipment
- I've added a few more fields to the query used on the details page, so you should be able to see the key data in that object!



## 💌 Any notes for the reviewer?

- I haven't added graphql layer test, as I'd heard from a couple of people that we were questioning the value of it ... should I still add it here?
- There's now an additional failing test, due to not yet implementing the sync translations.
